src/PluginsSystem/Tools/Microsoft.Performance.Toolkit.Plugins.Cli/Commands/Pack/PackCommand.cs

```diff
@@ -0,0 +1,80 @@
+﻿// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+using Microsoft.Extensions.Logging;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Processing;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Packaging;
+using Microsoft.Performance.Toolkit.Plugins.Runtime.Package;
+using Microsoft.Performance.Toolkit.Plugins.Cli.Manifest;
+
+namespace Microsoft.Performance.Toolkit.Plugins.Cli.Commands
+{
+    internal sealed class PackCommand
+        : ICommand<PackArgs>
+    {
+        private readonly IManifestLocatorFactory manifestLocatorFactory;
+        private readonly IPluginArtifactsProcessor sourceProcessor;
+        private readonly IPackageBuilder packageBuilder;
+        private readonly ILogger<PackCommand> logger;
+
+        public PackCommand(
+            IManifestLocatorFactory manifestLocatorFactory,
+            IPluginArtifactsProcessor sourceProcessor,
+            IPackageBuilder packageBuilder,
+            ILogger<PackCommand> logger)
+        {
+            this.manifestLocatorFactory = manifestLocatorFactory;
+            this.sourceProcessor = sourceProcessor;
+            this.packageBuilder = packageBuilder;
+            this.logger = logger;
+        }
+
+        public int Run(PackArgs args)
+        {
+            IManifestLocator manifestLocator = this.manifestLocatorFactory.Create(args);
```

Comment:
I'm noticing lines 34 through 46 are copy/pasted. Should we extract out a common dependency?

Reply:
Introduced a base command instead.